app/src/org/commcare/services/CommCareSessionService.java

```diff
@@ -438,6 +438,10 @@ public void closeServiceResources() {
         }
     }
 
+    public void invalidateUserDbHandle() {

```

Comment:
This is the result of a 412, right? I think we need to document exactly what state this creates for the current session and articulate what expectations should hold after it is executed.